src/Core/Contracts/Command.ts

```diff
@@ -1,38 +1,40 @@
 import Context from "./Context"
 
-export default abstract class Command {
-  public get description() {
-    return "Um comando que pode ser usado para interagir com o bot"
-  }
-
-  public get roles(): string[] {
-    return []
-  }
-
-  public get roleValidationMessages() {
-    return {}
-  }
-
-  public get validateAllRoles(): boolean {
-    return false
-  }
-
-  public get permissions(): string[] {
-    return []
-  }
-
-  public get permissionValidationMessages() {
-    return {}
-  }
-
-  public get validateAllPermissions(): boolean {
-    return false
-  }
-
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  public validate(context: Context): void | never {}
-
-  public abstract help(): string
-
-  public abstract async run(ctx: Context): Promise<void>
+interface Command {
+  readonly description: string
+  readonly roles: string[]
+  readonly roleValidationMessages: { [key: string]: string }
+  readonly validateAllRoles: boolean
+  readonly permissions: string[]
+  readonly permissionValidationMessages: { [key: string]: string }
+  readonly validateAllPermissions: boolean
+  readonly help: string
+  readonly validate: (context: Context) => Promise<void | never>
+  readonly run: (context: Context) => Promise<void>
+}
+type CommandOptions = Partial<Command> & {
+  help: Command["help"]
+  run: Command["run"]
 }
+const base = {
+  description: "Um comando que pode ser usado para interagir com o bot",
+  roles: [],
+  roleValidationMessages: [],
+  validateAllRoles: true, // TODO: should be true or false?
+  permissions: [],
+  permissionValidationMessages: {},
+  validateAllPermissions: true, // TODO: should be true or false?
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  validate: () => {},
+  help: undefined,
+  run: undefined,
+} as const
+
+const defaultObject = <T, U>(base: U, options: T): T & U =>
+  Object.fromEntries(
+    Object.keys(base).map((key) => [key, options[key] || base[key]])
+  ) as T & U
```

Comment:
This is essentially a lie, it only works because `base` has every key that `options` can have, as this is a "private" function that shouldn't be a problem